drf/open_questions.txt

```diff
@@ -0,0 +1,70 @@
+
+6. def my_view(request):
+    if request.method == 'GET':
+        return HttpResponse("This is a GET request")
+    elif request.method == 'POST':
+        return HttpResponse("This is a POST request")
+
+Serializers:
+7. the save() function in Django determines whether an object is new or existing based on the primary key value, and it handles the database operations accordingly. If the primary key is set (not None), it updates the existing record; otherwise, it creates a new one.
+8. errors = serializer.errors
+    print(errors)
+
+9. raise_exception=True is a way drf raise an exception if the data is not valid it will raise a serializers.ValidationError exception if the data is not valid.
+if not serializer.is_valid():
+        raise exceptions.ValidationError(serializer.errors)
+
+11. i will need to create a serializer for that one field and update the view accordingly.
+
+12. depth is the amount of levels you want to serialize inside the object, if i want the author data to be serialized and available to you i will set depth=1.
+
+14. Read-only fields are included in the API output, but should not be included in the input during create or update operations. Any 'read_only' fields that are incorrectly included in the serializer input will be ignored.
+
+15. SerializerMethodField is a versatile field that allows you to include custom data or calculated values in your serialized data. It is a read-only field, meaning that it can be used to display additional information that doesn't directly correspond to a model field
+
+Serializer Relations:
+
+16. In drf, a SlugRelatedField acts as a translator for related objects, enabling the representation of these objects using user-friendly slugs instead of numerical primary keys.
+if the read_only=False we need to make sure that unique=True
+
+17. select related is essentially a way to query sql join so we can get all the related objects data as well, pre-fetched related will get all related objects using more than one query that will be more useful in a many to many relationship,
+
+18. we need to create a Message serializer:
+
+from rest_framework import serializers
+from .models import Message
+
+class MessageSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Message
+        fields = ('id', 'content', 'timestamp', 'sender')
+
+and than in the user serializer we will create a field that will call the Message serializer:
+
+from rest_framework import serializers
+from .models import User
+from .serializers import MessageSerializer
+
+class UserSerializer(serializers.ModelSerializer):
+    messages = MessageSerializer(many=True, read_only=True)
```

Comment:
you could use what we just learned- SlugRelatedField, to show the messages

Reply:
ok,  but question didn't clarify to use it, i will use slugrelatedfield next time.